datalad/cmd.py

```diff
@@ -1327,9 +1328,27 @@ def close(self, return_stderr=False):
                 "Closing stdin of %s and waiting process to finish", process)
             process.stdin.close()
             process.stdout.close()
-            process.wait()
+            # try waiting for the annex process to finish 3 times for 3 sec
+            # with 1s pause in between
+            try:
+                try_multiple(
+                    # ntrials
+                    3,
+                    # exception to catch
+                    subprocess.TimeoutExpired,
+                    # base waiting period
+                    1.0,
+                    # function to run
+                    process.wait,
+                    timeout=3.0,
+                )
+            except subprocess.TimeoutExpired:
+                lgr.warning(
+                    "Batched process %s did not finish, abandoning it without killing it",
+                    process)
```

Comment:
should we just kill it at this point?  I am more afraid of run away git-annex processes lingering around and consuming increasing amount of resources in long running datalad processes

Reply:
I just had this discussion with @bpoldrack too (who feels like killing too). I am personally not in favor of such a move. The fact that these processes do not finish is a sign of an error. If we kill them, we create the impression that everything is splendid, but that is not really the case. Moreover, this is happening in `BatchedCommand`. There are no guarantees that the underlying process is a git-annex instance, that it wasn't supposed to write something to disk and is stalled by IO, that higher-level code would verify that all is kosher despite the process not having finished yet.

IMHO the only safe way to continue at this layer is to detach and leave these processes alone.